### PR TITLE
Merge latest reth changes from master

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

File: crates/consensus/authority/src/builder.rs [hunk 1]
```diff
@@ -65,7 +65,8 @@ pub enum AuthorityConsensusBuilderError {
 impl<Client, EvmConfig, Engine> AuthorityConsensusBuilder<Client, EvmConfig, Engine>
 where
     Engine: EngineTypes + 'static,
-    EvmConfig: ConfigureEvmEnv + Clone + Unpin + Send + Sync + 'static,
+    EvmConfig:
+        ConfigureEvmEnv + Clone + Unpin + Send + Sync + 'static + reth_node_api::ConfigureEvm,
```
Comment (arminsabouri) [inline crates/consensus/authority/src/builder.rs:69]: Do we need both `ConfigureEvmEnv` and `reth_node_api::ConfigureEvm`

**Review by arminsabouri — APPROVED** [review]
Looks good. Reviewed only the POA consensus things. Had one question

**Comment by arminsabouri** [comment]
Also I think its fine to create a merge commit for this PR. Rebasing will be messy as we have learned.

File: crates/consensus/authority/src/builder.rs (hunk 1 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/builder.rs:69]: We do, as they added a new trait which has the remaining functionality in it.

**Review by scottmillner — APPROVED** [review]

**Pull request merged**